mu.rb

```diff
@@ -26,13 +26,25 @@ def log
     @log
   end
 
-  def query(query)
+  def query(query, **options)
     log.info "Executing query: #{query}"
-    sparql_client.query query
+    @sparql_client = sparql_client(**options)
```

Comment:
Is there a specific reason to use an instance variable here?
Otherwise I would just do something like:
```ruby
authorized_sparql_client = sparql_client(**options)
authorized_sparql_client
```

Reply:
I think it's a remnant of trying out having a nil check for the sparql client instance variable, which I think allows re-using the client when there are multiple database calls within the same function.

A very old template version had the nil checks ( https://github.com/mu-semtech/mu-ruby-template/blob/v2.6.0-ruby2.3/sinatra_template/utils.rb#L37-L46 ), which is what the `mock-login-service` currently uses. Is there a reason current template versions don't have the check anymore?

Reply:
Wasn't aware that has changed over time. Don't think there is a specific reason. It's OK to keep it like this then :+1: